core/trino-main/src/main/java/io/trino/type/LikeFunctions.java

```diff
@@ -245,4 +246,14 @@ private static char getEscapeChar(Slice escape)
         }
         throw new TrinoException(INVALID_FUNCTION_ARGUMENT, "Escape string must be a single character");
     }
+
+    public static int getMatchingOffset(Matcher matcher, int at, int range)
+    {
+        try {
+            return matcher.matchInterruptible(at, range, Option.NONE);
+        }
+        catch (InterruptedException interruptedException) {
+            throw new TrinoException(GENERIC_USER_ERROR, "Regexp matching interrupted");
```

Comment:
Recover interrupted flag `Thread.currentThread().interrupt()`

Reply:
Hey, it is already done in the Joni's library:

  if (interruptCheckCounter++ % INTERRUPT_CHECK_EVERY == 0 && currentThread.isInterrupted()) {
               currentThread.interrupted();
               throw new InterruptedException();
 }

Reply:
Here is the java doc for the `Thread#interrupted` method:

```
Tests whether the current thread has been interrupted. The interrupted status of the thread is cleared by this method. In other words, if this method were to be called twice in succession, the second call would return false (unless the current thread were interrupted again, after the first call had cleared its interrupted status and before the second call had examined it).
```

The JONI library is compliant with the `InterruptedException` contract. They reset the `interrupted` flag before throwing an exception. Since the `InterruptedException` is being caught the interrupt flag must either be recovered or the thread must be terminated. Since you are simply throwing a different exception the interrupt flag must be recovered to propagate the interrupted status to the upper level code.

Reply:
I see.. okey, I can recover the thread interrupt flag.